### ISSUE TITLE
Accessibility problem for a visually impaired users

### ISSUE THREAD

**Issue by evgeniy-ADG** [AdguardTeam/AdguardForAndroid#519]
I use screen reader android : google talkback .
The application is very affordable, but has a problem of accessibility that needs to be solved .
In firewall with talkback I can not know if the option to block an application is enabled or disabled .
When it activates and deactivates what changes is the color of the icon and this information is not read by the talkback .
I hope this problem is resolved in the next update .

BMT-235-10270

**Comment by TPS** [comment]
I suppose this is as important as #151?

**Comment by ameshkov** [comment]
Basically, we should add "android:contentDescription" attribute

**Comment by Revertron** [comment]
Added content description 'Activated', 'Deactivated' and 'Disabled'.
Need translation of new strings.

**Comment by Bohdan-SUP** [comment]
User confirmed it's a lot better with the 2.6.23 beta.

**Comment by Bohdan-SUP** [comment]
User 1263433 reported Talkback is not recognizing widget on his MOTO X 2014 with Android 6.0

It works as intended on my HTC Desire 601 with Android 4.4.2

**Comment by Revertron** [comment]
Widget was enhanced in #695

**Comment by Bohdan-SUP** [comment]
Another issue here:
>As a JAWS screen reader user, I am unable to enter my license key to
activate AdGuard due to a pop up dialogue box which blocks the edit field.
Would it be possible for the devs to download a free demo version of JAWS
from the link below AND kindly address this issue.

http://www.freedomscientific.com/Downloads

**Comment by ameshkov** [comment]
@Bohdan-SUP let's move it to a separate issue tho

**Comment by Bohdan-SUP** [comment]
https://github.com/AdguardTeam/AdguardForAndroid/issues/1120
